### PR TITLE
generate local county mapping from CHNG spreadsheet

### PR DESCRIPTION
**Pull request by nmdefries**
### Description
In case CHNG makes changes to their county mappings, include the code that takes the Google spreadsheet, assigns group numbers, and saves locally. Be careful to keep from reusing group numbers within a state. Doing so would make historical data uninterpretable or difficult to use/document.

### Changelog
- `geo_data_proc.py`

### CODE REVIEW

File: _delphi_utils_python/data_proc/geomap/geo_data_proc.py
Comment (krivard) [inline _delphi_utils_python/data_proc/geomap/geo_data_proc.py:588]: suggestion: add tests for this or defer to when it's needed

I'm glad you thought about how to handle updating the group mapping, but the procedure is sufficiently subtle that I'd feel better about it if we had tests for it, and that would mean figuring out how to write and run tests for code that's not in the actual python module. That sounds like work that won't be needed for a while though, so we could also decide to leave handling the already-exists case aside until we actually have a group-change situation to handle: either drop the code from this PR or throw an exception so we don't run it by mistake.

File: _delphi_utils_python/data_proc/geomap/geo_data_proc.py
Comment (krivard) [inline _delphi_utils_python/data_proc/geomap/geo_data_proc.py:572]: question: are the `countyNN` columns ever used?

iirc those columns were ones I added for some analysis preprocessing, and didn't actually come from CHNG. if something downstream depends on them, we can retain the preprocessing step, but if not we may as well exclude them from the df

**Review by krivard — CHANGES_REQUESTED** [review]
nice refactor, but it might be _too_ nice -- we should resolve the already-exists handling before merging.

File: _delphi_utils_python/data_proc/geomap/geo_data_proc.py
Comment (nmdefries) [reply _delphi_utils_python/data_proc/geomap/geo_data_proc.py:572]: Oh, I didn't realize the `countyNN` fields weren't in the original spreadsheet. They're used to [turn the spreadsheet into long format](https://github.com/cmu-delphi/covidcast-indicators/blob/ndefries/geomapper/popsafe-county-level/_delphi_utils_python/data_proc/geomap/geo_data_proc.py#L495-L496), but I could also split `fips_list` if we don't want to rely on them being available.

Comment (dshemetov) [reply _delphi_utils_python/data_proc/geomap/geo_data_proc.py:572]: I got nerd sniped, so here's split trick, if you didn't get to it yet
```py
sheet_id = "1PEce4CjjHbRM1Z5xEMNI6Xsq_b2kkCh0"
sheet_name = "groupings"
# Request sheet in CSV format via tag in URL.
url = f"https://docs.google.com/spreadsheets/d/{sheet_id}/gviz/tq?tqx=out:csv&sheet={sheet_name}"

county_groups = pd.read_csv(
        url, dtype="string", index_col=False
    ).dropna(
        how="all", axis=1
    )
county_groups = county_groups[["popsafe-id", "state FIPS", "county FIPS grouping"]]
county_groups["state FIPS"] = county_groups["state FIPS"].astype(int)
county_groups = pd.concat([county_groups, county_groups["county FIPS grouping"].str.split("|", expand=True)], axis=1)
```

File: _delphi_utils_python/data_proc/geomap/geo_data_proc.py
Comment (dshemetov) [reply _delphi_utils_python/data_proc/geomap/geo_data_proc.py:588]: vote: keep the old+new groups merging code so we have a starting point if there's a change in the future, but by default throw an exception if `not county_groups.empty`. 

(agree with Katie: this code is hard to reason about without having a REPL and the data open, but unit testing this is probably not worth it, since a small change in their format could break our code)

**Review by dshemetov — CHANGES_REQUESTED** [review]
Second Katie's comment. Should be g2g after!

File: _delphi_utils_python/data_proc/geomap/geo_data_proc.py
Comment (dshemetov) [inline _delphi_utils_python/data_proc/geomap/geo_data_proc.py:610]:
```suggestion
                "code below is not tested or robust to changes in input format."
                "We recommend manually working with the code below and the new"
                "data in a REPL."
```

**Review by dshemetov — APPROVED** [review]
Minor nit and 🚀

**Comment by nmdefries** [comment]
@korlaxxalrok This is ready to merge.

**Pull request merged**
